### PR TITLE
Update PlayFabCommmonUtils.cpp.ejs

### PR DESCRIPTION
**Pull request by toddbellMSFT**
adding in a deprecated unreal call to just 4.20 logic paths.

### CODE REVIEW

File: targets/UnrealMarketplacePlugin/source/PlayFab/Source/PlayFabCommon/Private/PlayFabCommmonUtils.cpp.ejs [hunk 1]
```diff
@@ -1,9 +1,27 @@
-﻿<%- copyright %>
+<%- copyright %>
 
 #include "PlayFabCommonUtils.h"
 
 using namespace PlayFabCommon;
 
+// see {your unreal root folder}\Engine\Source\Runtime\Launch\Resources\Version.h for more preprocessor defines
+#if ENGINE_MINOR_VERSION==20
```
Comment (nima-ap) [inline targets/UnrealMarketplacePlugin/source/PlayFab/Source/PlayFabCommon/Private/PlayFabCommmonUtils.cpp.ejs]: Do higher version qualify for `==20` or should this be `>=20`?

Comment (nima-ap) [reply targets/UnrealMarketplacePlugin/source/PlayFab/Source/PlayFabCommon/Private/PlayFabCommmonUtils.cpp.ejs]: Nvm, I see that your goal was to add logic specifically for 4.20.

Comment (toddbellMSFT) [reply targets/UnrealMarketplacePlugin/source/PlayFab/Source/PlayFabCommon/Private/PlayFabCommmonUtils.cpp.ejs]: Nope, this version of GetEnvironmentVariable with 3 parameters is the only way to access the system variables in 4.20. For later versions (21 and 22) there's the single parameter version which is considered the "correct" version (3 params is labeled as deprecated http://api.unrealengine.com/INT/API/Runtime/Core/GenericPlatform/FGenericPlatformMisc/GetEnvironmentVariable/2/index.html)

Comment (pgilmorepf) [reply targets/UnrealMarketplacePlugin/source/PlayFab/Source/PlayFabCommon/Private/PlayFabCommmonUtils.cpp.ejs]: It's actually a fix for less than 4.20.
The current code works for 4.22, but went uncaught until we tried to verify on 4.20.

Comment (nima-ap) [reply targets/UnrealMarketplacePlugin/source/PlayFab/Source/PlayFabCommon/Private/PlayFabCommmonUtils.cpp.ejs]: Thanks for the fix

File: targets/UnrealMarketplacePlugin/source/PlayFab/Source/PlayFabCommon/Private/PlayFabCommmonUtils.cpp.ejs [hunk 2]
```diff
@@ -1,9 +1,27 @@
...
+FString PlayFabCommonUtils::GetTempDir()
+{
+    FString vars[] = { TEXT("TEMPDIR"), TEXT("TEMP"), TEXT("TMP") };
+    TCHAR FileSystemPath[256];
+    for (FString& envVar : vars)
+    {
+       FileSystemPath[0] = 0;
```
Comment (debhaldarMS) [inline targets/UnrealMarketplacePlugin/source/PlayFab/Source/PlayFabCommon/Private/PlayFabCommmonUtils.cpp.ejs]: Something weird about this formatting

File: targets/UnrealMarketplacePlugin/source/PlayFab/Source/PlayFabCommon/Private/PlayFabCommmonUtils.cpp.ejs [hunk 3]
```diff
@@ -1,9 +1,27 @@
...
 #include "PlayFabCommonUtils.h"
 
 using namespace PlayFabCommon;
 
+// see {your unreal root folder}\Engine\Source\Runtime\Launch\Resources\Version.h for more preprocessor defines
+#if ENGINE_MINOR_VERSION==20
+FString PlayFabCommonUtils::GetTempDir()
+{
+    FString vars[] = { TEXT("TEMPDIR"), TEXT("TEMP"), TEXT("TMP") };
+    TCHAR FileSystemPath[256];
```
Comment (debhaldarMS) [inline targets/UnrealMarketplacePlugin/source/PlayFab/Source/PlayFabCommon/Private/PlayFabCommmonUtils.cpp.ejs]: This is used in 3 places. Please define a local const and use it to futureproff out of sync issues

File: targets/UnrealMarketplacePlugin/source/PlayFab/Source/PlayFabCommon/Private/PlayFabCommmonUtils.cpp.ejs (hunk 1 above)
Comment (debhaldarMS) [inline targets/UnrealMarketplacePlugin/source/PlayFab/Source/PlayFabCommon/Private/PlayFabCommmonUtils.cpp.ejs]: Add some verbiage around why this is necessary and a code link to the Unreal Docs which shows this is deprecated

**Comment by toddbellMSFT** [comment]
retest

**Pull request merged**